clientcompat/internal/clientcompat/clientcompat.twirp.go

```diff
@@ -43,30 +43,32 @@ type CompatService interface {
 // =============================
 
 type compatServiceProtobufClient struct {
-	urlBase string
-	client  *http.Client
+	client *http.Client
+	urls   [2]string
 }
 
 // NewCompatServiceProtobufClient creates a Protobuf client that implements the CompatService interface.
 // It communicates using protobuf messages and can be configured with a custom http.Client.
 func NewCompatServiceProtobufClient(addr string, client *http.Client) CompatService {
+	prefix := urlBase(addr) + CompatServicePathPrefix
 	return &compatServiceProtobufClient{
-		urlBase: urlBase(addr),
-		client:  withoutRedirects(client),
+		client: withoutRedirects(client),
+		urls: [2]string{
+			prefix + "Method",
+			prefix + "NoopMethod",
+		},
 	}
 }
 
 func (c *compatServiceProtobufClient) Method(ctx context.Context, in *Req) (*Resp, error) {
-	url := c.urlBase + CompatServicePathPrefix + "Method"
 	out := new(Resp)
-	err := doProtoRequest(ctx, c.client, url, in, out)
+	err := doProtoRequest(ctx, c.client, c.urls[0], in, out)
```

Comment:
If these literal numbers are too opaque, they could be swapped for `iota` `const`s, or the whole array could be swapped for a flat list of `methodNameURL string` fields.